notify.js

```diff
@@ -139,16 +153,30 @@
             'requireInteraction': this.options.requireInteraction,
             'silent': this.options.silent,
             'closeOnClick': this.options.closeOnClick
-        });
+        };
 
-        if (!this.options.requireInteraction && this.options.timeout && !isNaN(this.options.timeout)) {
-            setTimeout(this.close.bind(this), this.options.timeout * 1000);
-        }
+        if (this.options.serviceWorker) {
+            var self = this;
+
+            this.options.serviceWorker.ready.then(function(registration) {
+                registration.showNotification(title, options)
+                .then(function() {
+                    return registration.getNotifications();
+                })
+                .then(function(notifications) {
+                    self.myNotify = notifications[notifications.length - 1]; // TODO: Naive implementation, improve
+                    bindNotificationEvents(self);
```

Comment:
but just by reading the MDN documentation that is not clear, since `NotificationEvent.notification` is documented as returning a regular `Notification` object which, in theory, should support the events...